app/controllers/items_controller.rb

```diff
@@ -1,19 +1,39 @@
 class ItemsController < ApplicationController
   before_action :authenticate_user!
-  before_action :set_wishlist_and_item, only: %i(edit update)
+  before_action :set_wishlist, only: %i(new create edit update)
+  before_action :set_item, only: %i(edit update)
+
+  def new
+    @item = @wishlist.items.new
+    authorize! :new, @item
+  rescue CanCan::AccessDenied
+    redirect_to @wishlist, alert: t("generic_error")
```

Comment:
Yeah, we should catch this error in one place only. Don't really know how to achieve this. Separate issue? I'd let it stay here the way it is for now.